src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -274,11 +286,35 @@ public static void sendExtras(Bundle extras) {
         }
     }
 
+	/*
+     * Retrives badge count from SharedPreferences
+     */
+	public static int getApplicationIconBadgeNumber(Context context){
+        SharedPreferences settings = context.getSharedPreferences("badge", Context.MODE_PRIVATE);
+        int badge = settings.getInt("badge", 0);
+        return badge;
+    }
+	
+	/*
+     * Sets badge count on application icon and in SharedPreferences
+     */
     public static void setApplicationIconBadgeNumber(Context context, int badgeCount) {
         if (badgeCount > 0) {
             ShortcutBadger.applyCount(context, badgeCount);
+
+			//This block is used to set the badge value in SharedPreferences key "badge", 
+            //in order for plugin cordova-plugin-badge to retrive it correctly
+			SharedPreferences.Editor editor = context.getSharedPreferences("badge", Context.MODE_PRIVATE).edit();
+			editor.putInt("badge", badgeCount);
+			editor.apply();
         } else {
             ShortcutBadger.removeCount(context);
+			
+			//This block is used to set the badge value in SharedPreferences key "badge", 
+            //in order for plugin cordova-plugin-badge to retrive it correctly
+			SharedPreferences.Editor editor = context.getSharedPreferences("badge", Context.MODE_PRIVATE).edit();
+			editor.putInt("badge", 0);
+			editor.apply();
```

Comment:
There is some duplicate code here in the if/else block. Why not move the setting of the badge number to after the if/else and do something like:

```
SharedPreferences.Editor editor = context.getSharedPreferences("badge", Context.MODE_PRIVATE).edit();
editor.putInt("badge", badge > 0 ? badge : 0);
editor.apply();
```